dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/visualization/DefaultVisualizationService.java

```diff
@@ -86,4 +91,34 @@ public Visualization getVisualizationNoAcl( String uid )
     {
         return visualizationStore.getByUidNoAcl( uid );
     }
+
+    @Override
+    @Transactional
+    public void removeExpressionDimensionItem( ExpressionDimensionItem expressionDimensionItem )
```

Comment:
I am wondering, should this delete related logic be put in a `DeletionHandler` instead? This is what we normally do for cleaning up associations. I think this code fits in `VisualizationDeletionHandler`.